### PR TITLE
Apply ruff

### PR DESCRIPTION
**Pull request by ndaelman-hu**

### CODE REVIEW

**Comment by ndaelman-hu** [comment]
So, 1 question is about the `pycodestyle` and `pylint` dependencies: can these dependencies be removed with `ruff` present?

**Comment by blueraft** [comment]
> So, 1 question is about the `pycodestyle` and `pylint` dependencies: can these dependencies be removed with `ruff` present?

Yes, they should be removed. You can remove the `pycodestyle` and `pylint` step in the workflow file too. 

Currently the workflow file also has a `ruff` lint check but no format check so you might want to add one for that too. This can be added the following way: 
https://github.com/nomad-coe/nomad-parser-example/blob/2bd4d49996b3d300c87d2225c5433c41e8eb7ce7/.github/workflows/actions.yaml#L49C1-L55C29

Folks at astral are working on having a unified command for both lint and format but it's not done yet - https://github.com/astral-sh/ruff/issues/8232.

File: pyproject.toml [hunk 1]
```diff
@@ -39,11 +39,11 @@ tests = [
     'phonopy==2.11.0',
     'h5py==3.6.0',
     'typing-extensions==4.4.0',
-    'ruff==0.1.4',
+    'ruff==0.1.8',
```
Comment (blueraft) [inline pyproject.toml:42]: Can v0.2.2 be used instead? I'm not sure of the editor integrations working in a submodule setup though.

Comment (ndaelman-hu) [reply pyproject.toml:42]: I can try, but I'd like to keep it the same to the main NOMAD\

Comment (blueraft) [reply pyproject.toml:42]: Yeah that's okay, you can leave it as is.

**Comment by ndaelman-hu** [comment]
> > So, 1 question is about the `pycodestyle` and `pylint` dependencies: can these dependencies be removed with `ruff` present?
> 
> Yes, they should be removed. You can remove the `pycodestyle` and `pylint` step in the workflow file too.
> 
> Currently the workflow file only has a `ruff` lint check but no format check so you might want to add one for that too. This can be added the following way: https://github.com/nomad-coe/nomad-parser-example/blob/2bd4d49996b3d300c87d2225c5433c41e8eb7ce7/.github/workflows/actions.yaml#L49C1-L55C29
> 
> Folks at astral are working on having a unified command for both lint and format but it's not done yet - [astral-sh/ruff#8232](https://github.com/astral-sh/ruff/issues/8232).

In the link you provided, ruff formatting is still commented out, though...

**Comment by blueraft** [comment]
> > > So, 1 question is about the `pycodestyle` and `pylint` dependencies: can these dependencies be removed with `ruff` present?
> > 
> > 
> > Yes, they should be removed. You can remove the `pycodestyle` and `pylint` step in the workflow file too.
> > Currently the workflow file only has a `ruff` lint check but no format check so you might want to add one for that too. This can be added the following way: https://github.com/nomad-coe/nomad-parser-example/blob/2bd4d49996b3d300c87d2225c5433c41e8eb7ce7/.github/workflows/actions.yaml#L49C1-L55C29
> > Folks at astral are working on having a unified command for both lint and format but it's not done yet - [astral-sh/ruff#8232](https://github.com/astral-sh/ruff/issues/8232).
> 
> In the link you provided, ruff formatting is still commented out, though...

Yeah 😅. Markus wanted auto-formatting to be optional in the example templates, but we are opting-in to auto-format code in this repo.

**Comment by ndaelman-hu** [comment]
@ladinesa If this fixes the pipeline, then we should consider adding merge checks again, i.e. you can only merge when passing the pipeline.
I would also split the build test from the python tests...

**Comment by blueraft** [comment]
LGTM from my side, you can also remove the `pylintrc` file.

**Comment by ndaelman-hu** [comment]
> LGTM from my side, you can also remove the `pylintrc` file.

Applied it in a separate merge.

**Pull request merged**